sklearn/calibration.py

```diff
@@ -887,6 +976,137 @@ def loss_grad(AB):
+        - If shape is (n_samples, 1), converts to (n_samples, 2) with (-x, x).
+        - If shape is (n_samples, n_classes), returns unchanged.
+        - For probability estimates, returns `numpy.log(decision_values + eps)`.
+
+    eps : float
+        Small positive value added to avoid log(0).
+
+    Returns
+    -------
+    logits : ndarray of shape (n_samples, n_classes)
+    """
+    if (decision_values.ndim == 2) and (decision_values.shape[1] > 1):
+        # Check if it is the output of `predict_proba`
+        entries_zero_to_one = np.all((decision_values >= 0) & (decision_values <= 1))
+        row_sums_to_one = np.all(np.isclose(np.sum(decision_values, axis=1), 1.0))
+
+        if entries_zero_to_one and row_sums_to_one:
+            logits = np.log(decision_values + eps)
+        else:
+            logits = decision_values
+
+    elif (decision_values.ndim == 2) and (decision_values.shape[1] == 1):
+        logits = np.hstack([-decision_values, decision_values])
+
+    elif decision_values.ndim == 1:
+        decision_values = decision_values.reshape(-1, 1)
+        logits = np.hstack([-decision_values, decision_values])
+
+    return logits.astype(dtype=decision_values.dtype)
+
+
+def _temperature_scaling(predictions, labels, sample_weight=None, beta_0=1.0):
+    """Probability Calibration with temperature scaling
+
+    Parameters
+    ----------
+    predictions : ndarray of shape (n_samples,) or (n_samples, n_classes)
+        The output of `decision_function` or `predict_proba`. If the input
+        appears to be probabilities (i.e., values between 0 and 1 that sum to 1
+        across classes), it will be converted to logits using `np.log(p + eps)`.
+
+        Binary decision function outputs (1D) will be converted to two-class
+        logits of the form (-x, x). For shapes of the form (n_samples, 1), the same
+        process applies.
+
+    labels : ndarray of shape (n_samples,)
+        True labels for the samples.
+
+    sample_weight : array-like of shape (n_samples,), default=None
+        Sample weights. If None, then samples are equally weighted.
+
+    beta_0 : float, default=1.0
+       Initial inverse temperature to start the optimisation.
+
+    Returns
+    -------
+    beta : float
+        The optimised inverse temperature parameter for probability calibration,
+        with a value in the range (0, infinity).
+
+    References
+    ----------
+    On Calibration of Modern Neural Networks,
+    C. Guo, G. Pleiss, Y. Sun, & K. Q. Weinberger, ICML 2017.
+    """
+    check_consistent_length(predictions, labels)
+    logits = _standardize_decision_values(predictions)
+
+    dtype_ = logits.dtype
+    labels = column_or_1d(labels, dtype=dtype_)
+
+    if sample_weight is not None:
+        sample_weight = _check_sample_weight(sample_weight, labels, dtype=dtype_)
+
+    halfmulti_loss = HalfMultinomialLoss(
+        sample_weight=sample_weight, n_classes=logits.shape[1]
+    )
+
+    def beta_loss(beta=beta_0):
```

Comment:
```suggestion
    def log_loss(log_beta=log_beta_0):
```

Reply:
So we purposely treat the original beta as log_beta to change the formulation of the optimization problem?

Reply:
Now I get what you mean. I changed the suggestion accordingly.